src/AspNetCoreModuleV2/InProcessRequestHandler/inprocessapplication.cpp

```diff
@@ -403,18 +403,19 @@ IN_PROCESS_APPLICATION::SetEnvironementVariablesOnWorkerProcess(
     {
         goto Finished;
     }
+    table.reset(pHashTable);
 
-    pHashTable->Apply(ENVIRONMENT_VAR_HELPERS::AppendEnvironmentVariables, &hr);
-    if (FAILED(hr))
-    {
-        goto Finished;
-    }
-    pHashTable->Apply(ENVIRONMENT_VAR_HELPERS::SetEnvironmentVariables, &hr);
+    table->Apply(ENVIRONMENT_VAR_HELPERS::AppendEnvironmentVariables, &hr);
     if (FAILED(hr))
     {
         goto Finished;
     }
+
+    table->Apply(ENVIRONMENT_VAR_HELPERS::SetEnvironmentVariables, &hr);
+
 Finished:
+    table->Clear();
```

Comment:
You can add a deleter that would clear table too.

Reply:
Yeah I'll do that.